metpy/plots/skewt.py

```diff
@@ -812,15 +812,17 @@ def plot_colormapped(self, u, v, c, bounds=None, colors=None, **kwargs):
         """
         u, v, c = delete_masked_points(u, v, c)
 
+        # Plotting a color segmented hodograph
         if colors:
             cmap = mcolors.ListedColormap(colors)
+            # If we are segmenting by height (a length), interpolate the bounds
             if bounds.dimensionality == {'[length]': 1.0}:
-                bounds = np.asarray(bounds + c[0]) * bounds.units
-                interp_vert = interp(bounds, c, c, u, v)
-                inds = np.searchsorted(c.magnitude, bounds.magnitude)
-                u = np.insert(u.magnitude, inds, interp_vert[1].magnitude)
-                v = np.insert(v.magnitude, inds, interp_vert[2].magnitude)
-                c = np.insert(c.magnitude, inds, interp_vert[0].magnitude)
+                bounds = bounds + c[0]  # Make all heights AGL
+                interpolation_heights = concatenate((bounds, c))
+                interpolation_heights = np.sort(interpolation_heights) * bounds.units
```

Comment:
Probably should be `interpolation_heights.units`. Not sure if we're guaranteed to get bounds units out.